accounts-db/src/ancient_append_vecs.rs

```diff
@@ -98,20 +98,21 @@ impl AncientSlotInfos {
         can_randomly_shrink: bool,
         ideal_size: NonZeroU64,
         is_high_slot: bool,
+        is_candidate_for_shrink: bool,
     ) -> bool {
         let mut was_randomly_shrunk = false;
         let alive_bytes = storage.alive_bytes() as u64;
         if alive_bytes > 0 {
             let capacity = storage.accounts.capacity();
             let should_shrink = if capacity > 0 {
-                let alive_ratio = alive_bytes * 100 / capacity;
-                alive_ratio < 90
-                    || if can_randomly_shrink && thread_rng().gen_range(0..10000) == 0 {
-                        was_randomly_shrunk = true;
-                        true
-                    } else {
-                        false
-                    }
+                if is_candidate_for_shrink {
+                    true
+                } else if can_randomly_shrink && thread_rng().gen_range(0..10000) == 0 {
```

Comment:
is this a change request or thinking aloud?

Reply:
what's wrong with using the is_candidate_for_shrink method? it encapsulates the logic

Reply:
why was the method added in the first place if using it is obtuse? or is just using it for shrinking ancient storages obtuse? So the change you want me to make is to code the condition in place, instead of calling the method and passing its result as a parameter to add(), correct?

Reply:
randomly shrinking doesn't matter anymore, now that we aren't including 55% of the # of ancient bytes that need to be shrunk. We're now including 0. So, random is not an issue, though it is confusing that it still remains.

Reply:
Yeah. That's another good thing about this PR - we now have *only one* place to determine shrinking by checking alive bytes ratio threshold.

So, in future, we just need to replace live_bytes with live_bytes exclude 0 lamport single ref accounts in one place ...